Sources/Orbit/Support/Components/TextField.swift

```diff
@@ -94,36 +117,94 @@ struct TextField: UIViewRepresentable {
             self.inputFieldEndEditingAction = inputFieldEndEditingAction
+            self.inputFieldEndEditingIdentifiableAction = inputFieldEndEditingIdentifiableAction
+            self.inputFieldShouldReturnAction = inputFieldShouldReturnAction
+            self.inputFieldShouldReturnIdentifiableAction = inputFieldShouldReturnIdentifiableAction
+            self.inputFieldShouldChangeCharactersAction = inputFieldShouldChangeCharactersAction
+            self.inputFieldShouldChangeCharactersIdentifiableAction = inputFieldShouldChangeCharactersIdentifiableAction
         }
 
         func textFieldDidBeginEditing(_ textField: UITextField) {
             inputFieldBeginEditingAction()
-        }
 
-        func textFieldDidEndEditing(_ textField: UITextField) {
-            if textField.isFirstResponder {
-                textField.resignFirstResponder()
+            if let identifier {
+                inputFieldBeginEditingIdentifiableAction(identifier)
             }
+        }
 
+        func textFieldDidEndEditing(_ textField: UITextField) {
             textFieldValue = textField.text ?? ""
+
             inputFieldEndEditingAction()
+
+            if let identifier {
+                inputFieldEndEditingIdentifiableAction(identifier)
+            }
         }
 
         func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-            textField.resignFirstResponder()
-            return true
+            let shouldReturn: Bool
+
+            if let inputFieldShouldReturnIdentifiableAction, let identifier {
+                shouldReturn = inputFieldShouldReturnIdentifiableAction(identifier)
+            } else if let inputFieldShouldReturnAction {
+                shouldReturn = inputFieldShouldReturnAction()
+            } else {
+                shouldReturn = true
+            }
+
+            if shouldReturn {
+                textField.resignFirstResponder()
+            }
+
+            return shouldReturn
         }
 
         func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
-            textFieldValue = ((textField.text ?? "") as NSString).replacingCharacters(in: range, with: string)
-            return true
+            let text = ((textField.text ?? "") as NSString)
+            let result: InputFieldShouldChangeResult
+
+            if let inputFieldShouldChangeCharactersIdentifiableAction, let identifier {
+                result = inputFieldShouldChangeCharactersIdentifiableAction(identifier, text, range, string)
+            } else if let inputFieldShouldChangeCharactersAction {
+                result = inputFieldShouldChangeCharactersAction(text, range, string)
+            } else {
+                result = .accept
+            }
+
+            switch result {
+                case .accept:
+                    // Accept the proposed change
+                    textFieldValue = text.replacingCharacters(in: range, with: string)
```

Comment:
Dispatch is required only for `.replace`, but not here?

Reply:
Oh, I guess not because nothing actually changes, rigth?

Reply:
Yep exactly - when we want the original UIKit handler to accept the changes, there is no need to do that. Only when we do need to change the value, we need to dispatch the update so it is called after rejecting the update.

Reply:
Updated in a separated commit. At the end, the dispatch is not needed - added different handling after fixing runtime warning issues.